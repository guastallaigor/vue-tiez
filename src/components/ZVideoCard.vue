<template>
  <div class="z-video-card" :class="{ dark }" :style="`width: ${width};`">
    <div v-if="src">
      <main>
        <div class="image">
          <figure>
            <main>
              <div class="container">
                <img :src="src" :alt="title || subtitle || ''">
              </div>
            </main>
          </figure>
        </div>
        <div class="play-icon">
          <svg width="23" height="30" viewBox="0 0 23 30" fill="none" xmlns="http://www.w3.org/2000/svg" aria-label="play">
            <path d="M23 14.875L0.5 29.4892L0.5 0.260821L23 14.875Z" fill="#fff"></path>
          </svg>
        </div>
      </main>
    </div>
    <div class="details" v-if="title && subtitle">
      <h4 :title="title" v-if="title">{{ title }}</h4>
      <p v-if="subtitle">{{ subtitle }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'z-video-card',
  props: {
    src: {
      type: String,
      required: true,
      default: ''
    },
    title: {
      type: String,
      default: ''
    },
    subtitle: {
      type: String,
      default: ''
    },
    width: {
      type: String,
      default: '500px'
    },
    dark: {
      type: Boolean,
      default: false
    }
  },
  data: () => ({})
}
</script>
