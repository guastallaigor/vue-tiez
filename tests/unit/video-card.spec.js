import { shallowMount } from '@vue/test-utils'
import ZVideoCard from '../../src/components/ZVideoCard'

describe('When I create the ZVideoCard component', () => {
  const createComponent = (propsData = {}, slot) => {
    const props = { ...propsData }
    props.src = props.src || 'https://image.mux.com/R6cYGw4NLjuquRZsxBuyXyzNLmIyZQjZ/thumbnail.jpg?time=15&width=430&height=240&smart_crop=true'

    return shallowMount(ZVideoCard, {
      propsData: props,
      slots: {
        default: slot || 'ZVideoCard'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent()
    expect(wrapper.find(ZVideoCard).isVueInstance()).toBe(true)
  })

  it('should have a default width of 500px', () => {
    const wrapper = createComponent()
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    expect(videoCard.classes().length).toBe(1)
    expect(videoCard.classes()).toEqual(['z-video-card'])
    expect(videoCard.attributes().style).toBe('width: 500px;')
  })

  it('should have a width of 666px', () => {
    const wrapper = createComponent({ width: '666px' })
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    expect(videoCard.classes().length).toBe(1)
    expect(videoCard.classes()).toEqual(['z-video-card'])
    expect(videoCard.attributes().style).toBe('width: 666px;')
  })

  it('should have an img src', () => {
    const wrapper = createComponent()
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    const img = wrapper.find('.z-video-card > div > main > .image > figure > main > .container > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().src).toBe('https://image.mux.com/R6cYGw4NLjuquRZsxBuyXyzNLmIyZQjZ/thumbnail.jpg?time=15&width=430&height=240&smart_crop=true')
  })

  it('should have an alt attribute and a title and subtitle', () => {
    const wrapper = createComponent({ title: 'Title', subtitle: 'Subtitle' })
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    const img = wrapper.find('.z-video-card > div > main > .image > figure > main > .container > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().alt).toBe('Title')
    const title = wrapper.find('.z-video-card > div + div > h4')
    expect(title.exists()).toBe(true)
    expect(title.text()).toBe('Title')
    const subtitle = wrapper.find('.z-video-card > div + div > p')
    expect(subtitle.exists()).toBe(true)
    expect(subtitle.text()).toBe('Subtitle')
  })

  it('should have an alt attribute with the title', () => {
    const wrapper = createComponent({ title: 'Title' })
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    const img = wrapper.find('.z-video-card > div > main > .image > figure > main > .container > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().alt).toBe('Title')
    const title = wrapper.find('.z-video-card > div + div > h4')
    expect(title.exists()).toBe(false)
    const subtitle = wrapper.find('.z-video-card > div + div > p')
    expect(subtitle.exists()).toBe(false)
  })

  it('should have an alt attribute with the subtitle', () => {
    const wrapper = createComponent({ subtitle: 'Subtitle' })
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    const img = wrapper.find('.z-video-card > div > main > .image > figure > main > .container > img')
    expect(img.exists()).toBe(true)
    expect(img.attributes().alt).toBe('Subtitle')
    const title = wrapper.find('.z-video-card > div + div > h4')
    expect(title.exists()).toBe(false)
    const subtitle = wrapper.find('.z-video-card > div + div > p')
    expect(subtitle.exists()).toBe(false)
  })

  it('should have a dark class', () => {
    const wrapper = createComponent({ dark: true })
    const videoCard = wrapper.find('.z-video-card')
    expect(videoCard.exists()).toBe(true)
    expect(videoCard.classes().length).toBe(2)
    expect(videoCard.classes()).toEqual(['z-video-card', 'dark'])
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      src: 'https://image.mux.com/R6cYGw4NLjuquRZsxBuyXyzNLmIyZQjZ/thumbnail.jpg?time=15&width=430&height=240&smart_crop=true',
      title: 'Title',
      subtitle: 'Subtitle',
      width: '500px',
      dark: false
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
