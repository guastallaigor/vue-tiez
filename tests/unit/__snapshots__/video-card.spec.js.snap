// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZVideoCard component should match snapshot 1`] = `
<div class="z-video-card" style="width: 500px;">
  <div>
    <main>
      <div class="image">
        <figure>
          <main>
            <div class="container"><img src="https://image.mux.com/R6cYGw4NLjuquRZsxBuyXyzNLmIyZQjZ/thumbnail.jpg?time=15&amp;width=430&amp;height=240&amp;smart_crop=true" alt="Title"></div>
          </main>
        </figure>
      </div>
      <div class="play-icon"><svg width="23" height="30" viewBox="0 0 23 30" fill="none" xmlns="http://www.w3.org/2000/svg" aria-label="play">
          <path d="M23 14.875L0.5 29.4892L0.5 0.260821L23 14.875Z" fill="#fff"></path>
        </svg></div>
    </main>
  </div>
  <div class="details">
    <h4 title="Title">Title</h4>
    <p>Subtitle</p>
  </div>
</div>
`;
